setup.py

```diff
@@ -34,59 +34,9 @@
 
 base_dir = path.abspath(path.dirname(__file__))
 
-doclines = __doc__.split('\n', 1)
```

Comment:
Should I remove the docsting in `setup.py`? It isn't used anymore for the project metadata.

Reply:
I see the long description has been replaced with a direct reference to README.rst, which is definitely nicer, and moving the single-line description directly into pyproject.toml seems fine. So, yes, I think getting rid of the doc string in setup.py makes sense. Is there any way to avoid having the file entirely, more like your last change? That would be preferable.

You managed to dynamically access version in the original commit. Couldn't something similar work for author & email? That should also eliminate the need to read & exec version.py. As for __url__, it looks like it may not be needed any in either setup.py or pyproject.toml, but for the sake of this change I'd leave __url__ in version.py as-is.

Reply:
> I think getting rid of the doc string in setup.py makes sense.

👍🏻 Done

> Is there any way to avoid having the file entirely, more like your last change? That would be preferable.
> 
> You managed to dynamically access version in the original commit. Couldn't something similar work for author & email? That should also eliminate the need to read & exec version.py.

Unfortunately, not that I know of. Setuptools only supports dynamic metadata for a few select fields.
https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata

The "best" alternative would probably be to either use `importlib.metadata` as suggested before or just duplicate the information. It doesn't really change so there wouldn't be much risk with both diverting.

> As for **url**, it looks like it may not be needed any in either setup.py or pyproject.toml, but for the sake of this change I'd leave **url** in version.py as-is.

`url` is actually used. Either for `Home-page` in `setup.py` or the `Project-URL: Homepage` in `pyproject.toml`. I can stay as is though for the time being.
```diff
-Home-page: http://asyncssh.timeheart.net
+Project-URL: Homepage, http://asyncssh.timeheart.net
```

Reply:
In that case, I think your original commit with setup.py deleted is probably the best option, but with the change in version.py reverted so all the variables remain set there. As you said, there's a small amount of duplication there, but that looks a lot cleaner overall.